survey/tests/views/test_survey_result.py

```diff
@@ -13,6 +13,13 @@ def test_survey_result(self):
         self.assertEqual(response.status_code, 200)
         response = self.client.get(reverse("survey-result", args=(1,)))
         self.assertEqual(response.status_code, 302)
+        status = self.client.get(reverse("survey-result", args=(4,))).status_code
+        self.assertEqual(status, 302)
+        if status == 302:
+            response = self.client.get(reverse("survey-result", args=(4,)), follow=True)
+            for message in response.context["messages"]:
+                msg = str(message)
```

Comment:
Maybe
```suggestion
                msg += str(message)
            self.assertIn("This survey has not been published", msg)
```
If there are multiple messages?

Reply:
This is a good point. I only assumed a single message. I was thinking something more along the lines of:

```
20            for message in response.context["messages"]:
21                if "published" in str(message):
22                    msg = str(message)
```
To filter for the message we are expecting. Or something similar? Seems to work fine but I have found that the messages are actually being duplicated. It went unnoticed before just because of the way I was grabbing it. I'll have to look into it more to see what is causing this.

I appreciate all the feedback. Will try to work on it more this week.

Reply:
The main thing I consider here is the amount of work to do if we add another message and we have to test it too. If we already have a string with all the messages, we just have to add another AssertIn for the other message. Also, AssertIn is less strict than assertEqual and would permit to change the current message a little (adding "Ask your survey admin to publish it."  at the end for example) without having to change the test. But your version would work too, I let you choose :)